packages/perseus/src/widgets/radio/base-radio.new.tsx

```diff
@@ -248,145 +249,155 @@ const BaseRadio = function ({
+                            });
+                        }
+
+                        const nextChoice = choices[i + 1];
+                        const nextChoiceHighlighted =
+                            // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+                            !!nextChoice && nextChoice.highlighted;
+
+                        const aphroditeClassName = (checked: boolean) => {
+                            // Whether or not to show correctness borders
+                            // for this choice and the next choice.
+                            return css(
+                                styles.item,
+                                styles.responsiveItem,
+                                checked && styles.selectedItem,
+                                checked &&
+                                    choice.highlighted &&
+                                    styles.aboveBackdrop,
+                                checked &&
+                                    choice.highlighted &&
+                                    apiOptions.isMobile &&
+                                    styles.aboveBackdropMobile,
+                                nextChoiceHighlighted &&
+                                    apiOptions.isMobile &&
+                                    styles.nextHighlighted,
+                            );
                         };
-                    }
-
-                    // TODO(mattdr): Index isn't a *good* choice of key
-                    // here; is there a better one? Can we use choice
-                    // content somehow? Would changing our choice of key
-                    // somehow break something happening inside a choice's
-                    // child Renderers, by changing when we mount/unmount?
-                    return (
-                        // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions -- TODO(LEMS-2871): Address a11y error
-                        <li
-                            key={i}
-                            ref={(e) => (listElem = e)}
-                            className={className}
-                            onClick={clickHandler}
-                            onTouchStart={
-                                labelWrap
-                                    ? undefined
-                                    : captureScratchpadTouchStart
-                            }
-                        >
-                            <Element {...elementProps} ref={ref} />
-                        </li>
-                    );
-                })}
-            </ul>
+
+                        // HACK(abdulrahman): Preloads the selection-state
+                        // css because of a bug that causes iOS to lag
+                        // when selecting the button for the first time.
+                        aphroditeClassName(true);
```

Comment:
I wonder if this is even necessary anymore. No need to change this as part of your PR though, just musing. :)